File: DesignPatterns/01_GOF23/DP_examples/Rust/01_Creational/Builder/builder.rs
pub trait IBuilder {
    fn build(&self) -> Box<dyn Item>;
}

pub trait IPacker {
    fn pack(&self, item: &dyn Item);
}

pub trait Item {
    fn name(&self) -> &str;
    fn price(&self) -> f32;
    fn packer(&self) -> &dyn IPacker;
}

pub enum ItemKind {
    VegetableBurger,
    ChickenBurger,
    CokeCola,
    PepsiCola,
}

File: DesignPatterns/01_GOF23/DP_examples/CSharp/01_Creational/Builder/Builder.cs
namespace Creational;
public interface IMealBuilder {
    List<IBuilder> GetBuilders();
}
public interface IBuilder {
    Item Build();
}
public interface IPacker {
    void Pack(Item item);
}
public interface Item {
    string name { get; }
    float price { get; }
    IPacker Packer { get; }
}